xla/service/gpu/cudnn_norm_rewriter.cc

```diff
@@ -171,6 +172,42 @@ bool CompatibleElementType(const HloInstruction* instr) {
   return element_type == BF16 || element_type == F16 || element_type == F32;
 }
 
+// Returns the dimensions associated with shape, adjusted for the removal of any
+// degenerate dimensions in shape.
```

Comment:
Maybe add the sentences:

Specifically, for each dimension 'd' in dimensions, returns the new index of d if all dimensions of size 1 are removed from 'shape'. If d itself has size 1, it is not included in the returned vector.

Reply:
This amendment greatly improved my understanding of the function without needing to look at the actual implementation of the function.